### DISCUSSION TITLE
SSL\TLS usage

### DISCUSSION THREAD

**Discussion by nazarii-nice** [Havret/dotnet-activemq-artemis-client#474] [Q&A] [answered]
Hi, how i can use a TLS?

**Comment by Havret** [comment, marked as answer]
Hi @nazar4k,

Here's a quick guide to setting up SSL/TLS in ArtemisNetClient:

First, create an endpoint with SSL/TLS enabled by setting the scheme to Scheme.Amqps:

```csharp
var endpoint = Endpoint.Create("localhost", 5672, "guest", "guest", scheme: Scheme.Amqps);
```

Next, configure your ConnectionFactory with these SSL settings (feel free to adjust them as needed):

```csharp
var connectionFactory = new ConnectionFactory
{
    SSL =
    {
        ClientCertificates = null,
        Protocols = SslProtocols.Tls13,
        CheckCertificateRevocation = false,
        RemoteCertificateValidationCallback = null,
        LocalCertificateSelectionCallback = null
    }
};
```

This sets up the SSL properties of the connection factory, including protocols and certificate settings.

Finally, create a connection and a producer to send messages:

```csharp
var connection = await connectionFactory.CreateAsync(endpoint);
var producer = await connection.CreateProducerAsync("my-address", RoutingType.Multicast);
await producer.SendAsync(new Message("foo"));
```

I hope that helps,
Havret

**Reply by nazarii-nice** [reply]
Thank you very much

**Reply by nazarii-nice** [reply]
Do I need to change something in the Artemis configuration to use the amqps scheme?

**Reply by Havret** [reply]
Yes, please take a look at the Artemis [docs](https://activemq.apache.org/components/artemis/documentation/latest/configuring-transports.html#configuring-netty-ssl) and the example [configuration](https://github.com/apache/activemq-artemis-examples/tree/main/examples/features/standard/ssl-enabled).

**Reply by nazarii-nice** [reply]
Do you know if I can use the .pfx certificate in keystorepath?

**Reply by nazarii-nice** [reply]
What I`m doing wrong?
I have only one acceptor   
<acceptor name="netty">tcp://0.0.0.0:61617?sslEnabled=true;keyStorePath=path/name.pfx;keyStorePassword=password</acceptor>

And the client
            string path = "path\\name.pfx";
            string password = "password";
            var cer = new X509Certificate(path, password);
            var cl = new X509CertificateCollection
            {
                cer
            };
        var endpoint = Endpoint.Create("fqdn", 61617, "user", "password", Scheme.Amqps);

        var connectionFactory = new ConnectionFactory()
        {
            SSL =
            {
                ClientCertificates = cl,
                Protocols = SslProtocols.Tls,
                CheckCertificateRevocation = false,
                RemoteCertificateValidationCallback = null,
                LocalCertificateSelectionCallback = null
            }
        };
        var connection = await connectionFactory.CreateAsync(endpoint);
        var producer = await connection.CreateProducerAsync("Test Address", RoutingType.Multicast);

**Reply by Havret** [reply]
I'm sorry, but I won't be able to assist with this matter. You'll need to reach out to the Artemis team. The best way to do so would be through their [mailing list](https://activemq.apache.org/contact).
